surface_potential_analysis/util/plot.py

```diff
@@ -319,7 +322,8 @@ def plot_data_2d(
     mesh.set_norm(norm)
     mesh.set_clim(*clim)
     ax.set_aspect("equal", adjustable="box")
-    fig.colorbar(mesh, ax=ax, format="%4.1e")
+    if get_colour_bar:
```

Comment:
instead of this, I think it would be better to do
```
def _has_colorbar(axis: Axes) -> bool:
    for artist in axis.get_children():
        if isinstance(artist, plt.cm.ScalarMappable) and artist.colorbar is not None:
            return True
    return False
```
and replace get_colorbar with
```
if not _has_colorbar(ax):
```